CONTRIBUTING.md

```diff
@@ -0,0 +1,234 @@
+
+### Development process
+
+From a process perspective, it is recommended to develop your desired end state first, which can then be reverse engineered back into a plugin/generators. Here is an example of the process followed for creating our `@ensono-stacks/cypress` plugin:
+
+***Step 1: Creating your end state***
+
+- Create a baseline workspace with a Next.js application. Commit this to your `main` branch
+
+- Create a new branch, e.g. `cypress-baseline` for the *init* generator. In this branch, install and configure the Cypress testing framework with the required configurations.
+
+- Create a new branch from `cypress-baseline` for the *init-deployment* generator (`cypress-deployment`), adding the additional requirements for deploying your application with cypress tests.
+
+- Create a new branch from `cypress-baseline` for the *accessibility* generator (`cypress-accessibility`), adding the additional requirements for adding accessibility tests using cypress to your application.
+
+***Step 2: Create your plugin, generators and unit tests.***
+
+TDD should be followed. For each generator to create you should write all of the unit tests to verify that the files you create or amend match the desired states defined in step 1.
+Following the approach outlined in step 1, tailored to your own plugin, helps to determine the behaviour and responsibility of each required generator. Through the git comparison functionality you gain a visual representation of exactly what has changed and in which files, making development of your unit tests MUCH easier!
+
+- Create a new plugin with the following command:
+
+```bash
+
+nx g @nx/plugin:plugin <plugin name> --importPath @ensono-stacks/<plugin name>
+
+```
+
+For each new generator, the following command can be run:
+
+```bash
+
+nx generate @nx/plugin:generator <generator name> --project=<plugin name>
+
+```
+
+- *init* generator: Use Git to review the differences between the `cypress-baseline` branch and the `main` branch. This will help identify the files your generator needs to manipulate or create, then write all of the unit tests to check that the desired end state is met.
+
+- *init-deployment* generator: Use Git to review the differences between the `cypress-deployment` branch and the `cypress-baseline` branch. This will help identify files which need to be amended/created on top of the *init* files and create your unit tests.
+
+- *accessibility* generator: Use Git to review the differences between the `cypress-accessibility` branch and the `cypress-baseline` branch. Again creating the relevant unit tests.
+
+***Step 3: Implementation***
+
+It is worth reviewing existing plugins to understand how we utilise the `@nx/devkit` and `@ensono-stacks/core` to create and manipulate files.
+
+Some important initial checks:
+
+1. `verifyPluginCanBeInstalled()`: Checks to see if the workspace is compatible

```

Comment:
is it worth mentioning that the these 2 functions (verifyPluginCanBeInstalled, hasGeneratorExecutedForProject) come from the @ensono-stacks/core package and should be called inside any generator for validation purposes.

Reply:
Updated, please review